python/raydp/spark/dataset.py

```diff
@@ -449,3 +455,59 @@ def create_ml_dataset_from_spark(df: sql.DataFrame,
                                  node_hints: List[str] = None) -> MLDataset:
     return RayMLDataset.from_spark(
         df, num_shards, shuffle, shuffle_seed, fs_directory, compression, node_hints)
+
+def spark_dataframe_to_ray_dataset(df: sql.DataFrame):
+    blocks, _ = _save_spark_df_to_object_store(df)
+    return from_arrow(blocks)
+
+def _convert_blocks_to_dataframe(blocks):
+    # connect to ray
+    if not ray.is_initialized():
+        # raise log level to avoid filling log with this
+        ray.init(address="auto", logging_level=logging.WARN)
+    for block in blocks:
+        dfs = []
+        for b in block["ref"]:
+            ref = ray.cloudpickle.loads(b)
+            data = ray.get(ref)
+            dfs.append(data.to_pandas())
+        yield pd.concat(dfs)
+
+def _convert_by_udf(spark: sql.SparkSession,
+                    blocks: List[ObjectRef],
+                    schema: StructType) -> DataFrame:
+    s = StructType([StructField("ref", BinaryType(), False)])
+    ref_list = [(ray.cloudpickle.dumps(block),) for block in blocks]
+    blocks_df = spark.createDataFrame(ref_list, schema)
```

Comment:
can we first create a RDD for ref_list here by solving the locality issue and specifying the correct partition number?